plugin/role_set.go

```diff
@@ -166,6 +169,32 @@ func (b *backend) saveRoleSetWithNewAccount(ctx context.Context, req *logical.Re
 		return nil, err
 	}
 
+	// Service accounts in GCP are eventually consistent, and can take over 60s
+	// to be ready for use. Attempt to GET the service account in a retry loop
+	// to ensure the service account is ready for use before continuing on
+	// and creating IAM bindings.
+	_, err = retryWithExponentialBackoff(ctx, func() (interface{}, bool, error) {
+		iamAdmin, err := b.IAMAdminClient(req.Storage)
+		if err != nil {
+			return nil, false, err
+		}
+		gcpAcct, err := b.getServiceAccount(iamAdmin, &gcputil.ServiceAccountId{
+			Project:   sa.ProjectId,
+			EmailOrId: sa.Email,
+		})
+
+		// Propagation delays within GCP can cause this error occasionally, so don't quit on it.
+		if err != nil && (strings.Contains(err.Error(), errDoesNotExist)) {
+			return nil, false, nil
+		}
+
+		return gcpAcct, true, err
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("error getting service account after creation: %w", err)
+	}
+
 	// Create new IAM bindings.
 	if err := b.createIamBindings(ctx, req, sa.Email, newResources.bindings); err != nil {
```

Comment:
Is it better to just retry this function instead? Or does it make more sense to ensure a service account can be retrieved?